services/fileServices/MdPageServices/__tests__/ContactUsPageService.spec.js

```diff
@@ -0,0 +1,171 @@
+        content: { frontMatter: mockFrontMatter, pageBody: mockContent },
+        sha: mockContactUsSha,
+      })
+
+      expect(retrieveDataFromMarkdown).toHaveBeenCalledWith(
+        mockRawContactUsContent
+      )
+      expect(mockGithubService.read).toHaveBeenCalledWith(reqDetails, {
+        fileName: CONTACT_US_FILE_NAME,
+        directoryName: CONTACT_US_DIRECTORY_NAME,
+      })
+      expect(mockFooterYmlService.read).toHaveBeenCalledWith(reqDetails)
+    })
+
+    it("Propagates the correct error on failed retrieval", async () => {
+      mockFooterYmlService.read.mockRejectedValueOnce(new NotFoundError(""))
+
+      await expect(service.read(reqDetails)).rejects.toThrowError(NotFoundError)
+
+      expect(retrieveDataFromMarkdown).toHaveBeenCalledWith(
+        mockRawContactUsContent
+      )
+      expect(mockGithubService.read).toHaveBeenCalledWith(reqDetails, {
+        fileName: CONTACT_US_FILE_NAME,
+        directoryName: CONTACT_US_DIRECTORY_NAME,
+      })
+      expect(mockFooterYmlService.read).toHaveBeenCalledWith(reqDetails)
+    })
+  })
+
+  describe("Update", () => {
+    const oldSha = "54321"
+    const updatedFeedback = "updated"
+    mockGithubService.update.mockResolvedValue({ newSha: mockContactUsSha })
+    it("Updating page content works correctly", async () => {
+      const mockUpdatedFrontMatter = {
+        ...mockContactUsContent.frontMatter,
+        feedback: updatedFeedback,
+      }
+
+      await expect(
+        service.update(reqDetails, {
+          fileName: CONTACT_US_FILE_NAME,
+          content: mockContent,
+          frontMatter: mockUpdatedFrontMatter,
+          sha: oldSha,
+        })
+      ).resolves.toMatchObject({
+        content: { frontMatter: mockUpdatedFrontMatter, pageBody: mockContent },
+        oldSha,
+        newSha: mockContactUsSha,
+      })
+
+      expect(convertDataToMarkdown).toHaveBeenCalledWith(
+        mockUpdatedFrontMatter,
+        mockContent
+      )
+      expect(mockGithubService.update).toHaveBeenCalledWith(reqDetails, {
+        fileName: CONTACT_US_FILE_NAME,
+        directoryName: CONTACT_US_DIRECTORY_NAME,
+        fileContent: mockRawContactUsContent,
+        sha: oldSha,
+      })
+      expect(mockFooterYmlService.read).toHaveBeenCalledWith(reqDetails)
+      expect(mockFooterYmlService.update).toHaveBeenCalledWith(reqDetails, {
+        fileContent: {
+          ...mockFooterContent,
+          feedback: updatedFeedback,
+        },
+        sha: mockFooterSha,
+      })
+    })
+    it("Propagates the correct error on failed update", async () => {
+      mockGithubService.update.mockRejectedValueOnce(new NotFoundError(""))
+
+      await expect(
+        service.update(reqDetails, {
+          fileName: CONTACT_US_FILE_NAME,
+          content: mockContent,
+          frontMatter: mockFrontMatter,
+          sha: oldSha,
+        })
+      ).rejects.toThrowError(NotFoundError)
```

Comment:
same comment as above, i think the readability might be better if we separated the action/expectation!